### ISSUE TITLE
Component organisation

### ISSUE THREAD

**Issue by peteryates** [DFE-Digital/govuk-components#2]
Currently (as of #1) the two example components are separated by directory

```
app/components
├── panel
│   ├── panel_component.html.erb
│   └── panel_component.rb
└── start_now_button
    ├── start_now_button_component.html.erb
    └── start_now_button_component.rb
```

And invoked like this:

```ruby
StartNowButton::StartNowButtonComponent.new(text: text, href: href)
```

This is ugly and repetitive. It would be nicer if they were namespaced under `GOVUK` and in a flatter structure:

```ruby
GOVUK::StartNowButtonComponent.new(text: text, href: href)
```

----

### Questions

* [x] Do we _need_ to name them all `<thing>Component`? Is the common suffix just adding noise?
* [ ] Are people ok with not separating components by directory? We'll probably have quite a few components and having them - the ruby files and templates - all in a single directory might get a bit _unwieldy_

**Comment by tvararu** [comment]
I like separating components by directory; I'd argue for colocating specs next to components as well.

👍  for GOVUK::StartNowButtonComponent

**Comment by tijmenb** [comment]
I think `GOVUK` might be too generic, when I was on GOV.UK we tried to avoid it because of the risk of collisions with internal & external libraries (compounded by problems with `Govuk` vs `GOVUK` and Rails inflection). 

To the suffix question, I think `-Component` is indeed noise if we have a namespace, so perhaps this works:

```rb
GovukComponent::Panel.new
GovukComponent::Tag.new
GovukComponent::StartNowButton.new
```

**Comment by peteryates** [comment]
I've submitted #5 and it feels like the the right approach. I can't see a way of keeping the names simple and predictable _and_ organising with one component per directory.

Leaving the task unchecked in case I've missed something obvious.

**Comment by peteryates** [comment]
Merged #5, I like this approach apart from the previously stated reasons (multiple components in one directory, the weird capitalisation of `Govuk` :disappointed:).

Everything else is better though, so until something better comes along let's go with it
